bin/pycbc_compress_bank

```diff
@@ -0,0 +1,172 @@
+parser.add_argument("--tmplt-index", nargs=2, type=int, default=None,
+                help="Only generate compressed waveforms for the given "
+                     "indices in the bank file. Must provide both a start "
+                     "and a stop index. Default is to compress all of the "
+                     "templates in the bank file.")
+parser.add_argument("--compression-algorithm", type=str, required=True,
+                choices=compress.compression_algorithms.keys(),
+                help="The compression algorithm to use for selecting "
+                     "frequency points.")
+parser.add_argument("--t-pad", type=float, default=0.001,
+                help="The minimum duration used for t(f) in seconds. The "
+                     "inverse of this gives the maximum frequency step that "
+                     "will be used in the compressed waveforms. Default is "
+                     "0.001.")
+parser.add_argument("--precision", type=float, default=0.001,
+                help="The maximum mismatch to allow between the interpolated "
+                     "waveform must and the full waveform. Points will be "
+                     "added to the compressed waveform until its "
+                     "interpolation has a mismatch <= this value. Default is "
+                     "0.001.")
+parser.add_argument("--interpolation", type=str, default="linear",
+                help="The interpolation to use for decompressing the "
+                     "waveforms for checking precision. Options are 'linear', "
+                     "or any interpolation recognized by scipy's interp1d "
+                     "kind argument. Default is linear.")
+parser.add_argument("--verbose", action="store_true", default=False)
+
+
+args = parser.parse_args()
+
+pycbc.init_logging(args.verbose)
+
+fmin = args.low_frequency_cutoff
+fmax = args.sample_rate/2.
+
+# load the bank
+logging.info("loading bank")
+# we'll just use dummy values for N, df for now
+bank = waveform.FilterBank(args.bank_file, 5, 0.25, fmin, numpy.complex128,
+    approximant=args.approximant)
+templates = bank.table
+if args.tmplt_index is not None:
+    imin, imax = args.tmplt_index
+    templates = templates[imin:imax]
+else:
+    imin, imax = 0, templates.size
+
+# figure out the dfs needed for each waveform
+logging.info("getting needed dfs")
+# we'll ensure that the there are atleast 2 samples in a waveform
+seg_lens = 4*numpy.array([max(4./args.sample_rate,
+    2**numpy.ceil(numpy.log2(compress.rough_time_estimate(m1, m2, fmin))))
+    for m1,m2 in zip(templates.mass1, templates.mass2)])
+
+# generate output file
+# FIXME: this should be moved to a class or function
+logging.info("writing template info to output")
+output = h5py.File(args.output, 'w')
+output['mass1'] = templates['mass1']
+output['mass2'] = templates['mass2']
+output['spin1z'] = templates['spin1z']
+output['spin2z'] = templates['spin2z']
+output['template_hash'] = templates['template_hash']
```

Comment:
Ian, Collin and I have talked about this and the next patch, will change how this is handled, and move the hashing functionality (and what columns it uses) into the TemplateBank class, so this can be kept consistent.